File: heex_ros/Heex_SDK_2_39_1/samples/python/basicSampleMonitor_Interval.py
#!/usr/bin/env python3
#
# Copyright (c) 2024 Heex Technologies
# All Rights Reserved.
#
# This file is subject to the terms and conditions defined in
# file 'LICENSE', which is part of this source code package.
#
import sys
import time
import random

# Adding the PyHeexCustomerSide Lib to PYTHONPATH. Depending from where you run this sample, you might need to update it to point to the lib
sys.path.insert(1, "../sdk/lib")
sys.path.insert(1, "../../sdk/lib")

from PyHeexCustomerSide import IntervalMonitor
from PyHeexCustomerSide import __version__ as HEEX_SYSTEM_SDK_VERSION


# This basic IntervalMonitor Sample creates a sample_interval_monitor instance. It'll then update the instance's humidity percentage it
# receives with random values, and if the percentage is within one of the provided intervals (provided during the creation
# of the monitor on the platform), the monitor activates.
if __name__ == '__main__':
    try:
        print(f"Heex SDK version : {HEEX_SYSTEM_SDK_VERSION}")
        # TODO: Replace the Monitor UUID with the one provided (by the Web Platform) for your Trigger or give a UUID as an input to the generated executable
        monitor_uuid = "M-XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX(1.0.0)"
        if len(sys.argv) > 1:
            monitor_uuid = sys.argv[1]

        server_ip = "127.0.0.1"
        server_port = 4242

        print(f"{sys.argv[0]} will be exposed as {monitor_uuid} tries to connect to {server_ip}:{server_port}.")

        # configure and launch Monitor
        sample_interval_monitor = IntervalMonitor(monitor_uuid, server_ip, server_port)
        sample_interval_monitor.setSignalUnit("%")  # Provide the unit of input data
        sample_interval_monitor.awaitReady()        # awaits the connection with the kernel.
        while True:
            humidity = random.randint(0, 100000) / 1000  # random humidity between 0 and 100 %
            print(f"Humidity is at {humidity} {sample_interval_monitor.getSignalUnit()}")
            sample_interval_monitor.updateValue(humidity)
            time.sleep(1)

    except Exception as e:
        print("Error", e)
        exit(1)

File: heex_ros/Heex_SDK_2_39_1/samples/cpp/basicSampleMonitor_Interval.cpp
///
/// Copyright (c) 2023 Heex Technologies
/// All Rights Reserved.
///
/// This file is subject to the terms and conditions defined in
/// file 'LICENSE', which is part of this source code package.

#include <fstream>
#include <iostream>
#include <string>
#include <thread> // std::this_thread::sleep_for

#include "HeexUtilsLog.h"
#include "IntervalMonitor.h"

// This basic IntervalMonitor Sample creates a sampleIntervalMonitor instance. It'll then update the instance's humidity percentage it
// receives with random values, and if the percentage is within one of the provided intervals (provided during the creation
// of the monitor on the platform), the monitor activates.
int main(int argc, char** argv)
{
  HEEX_LOG(info) << "Heex SDK version : " << Heex::HEEX_SYSTEM_SDK_VERSION << std::endl;

  std::srand(static_cast<unsigned int>(std::time(nullptr))); // Initialize random seed

  // TODO: Replace the  Monitor UUID with the one provided (by the Web Platform) for your Trigger or give a UUID as an input to the generated executable
  std::string monitorUuid = "M-XXXXXXXX-XXXX-XXXX-XXXX-XXXXXXXXXXXX(1.0.0)";
  if (argc > 1) // Parse the UUID as an optional parameter.
  {
    monitorUuid = argv[1];
  }
  const std::string serverIp    = "127.0.0.1";
  const unsigned int serverPort = 4242;

  HEEX_LOG(info) << argv[0] << " will be exposed as " << monitorUuid << " tries to connect to " << serverIp << ":" << serverPort << "." << std::endl;

  // configure and launch the Monitor
  IntervalMonitor sampleIntervalMonitor(monitorUuid, serverIp, serverPort);
  sampleIntervalMonitor.setSignalUnit("%"); // Provide the unit of input data
  sampleIntervalMonitor.awaitReady();       // awaits the connection with the kernel.

  while (true)
  {
    const double humidity = (std::rand() % 100000) / 1000.0; // random humidity between 0 and 100 %
    HEEX_LOG(info) << "Humidity is at " << humidity << " " << sampleIntervalMonitor.getSignalUnit() << std::endl;
    sampleIntervalMonitor.updateValue(humidity);
    std::this_thread::sleep_for(std::chrono::milliseconds(1000));
  }

  return EXIT_SUCCESS;
}
